### ISSUE TITLE
Error while restoring a save on mobile

### ISSUE THREAD

**Issue by AbsoluteJuicer** [Nuku/Flexible-Survival#2641]
I've been playing Flexible Survival on mobile for a long time and it was fine. I started encountering this problem since ~2019 (I forgot) and only recently have I started to familiarize with Github to report this issue. 

**iOS App Name and Version:** Frotz (1.8.8) (iOS versions did not prevent the problem from occurring, earlier Frotz versions also have this problem)
**Story File Version:** August 2023 (earlier versions also had this problem, unless it's really early, like 2016 versions)
**Error Message:** `***fatal error: Heap block summary is out of order.***
***fatal error: Heap block summary is out of order.***` (note: this message is repeated as shown)

**How to crash the game:** 
In short: Restoring a save will trigger it (click [Restore] -> click a savefile -> then the game will crash, showing the error message)
Full method: (start a new game) -> when you are in the bunker, click [import progress] -> after the progress is successfully imported, save game by clicking [save] -> now anytime the game will crash when restoring a save, while undoing is fine
